_posts/2024-09-05-releasing-apps.md

```diff
@@ -0,0 +1,85 @@
+
+Developers do their work in small, manageable chunks, and merge into a trunk branch: `main`. Every week (or fortnight in our case), a workflow kicks off to cut a _release branch_. This release branch behaves similarly to the GitFlow release branch—it’s fixing our base at a given point, allowing us to stabilize before releasing. Fixes for the release could go one of two ways - based on release and merged into the release branch, or based on `main`, and back-ported to the release branch.
+
+Merging a release branch back in to where it came from was one of the time sinks we were looking to get rid of, removing the need to resolve merge conflicts. Instead, we aimed for fixing on `main`, and _never_ merging the release branch back in.
+
+![image](/assets/img/2024-09-05-trunk-ish-branching.png)
+*Trunk branching with release branches*
+
+To make this “back-porting” simple for developers, we introduced an automatic cherry-picking workflow. When bugs come up in a release before deployment, pull requests with fixes would land in `main`, first. These pull requests have a special label which another piece of automation picks up on, knowing to cherry-pick the newly merged work on to the open release branch!
+
+This works the majority of the time, but what about merge conflicts? In our old school GitFlow model, resolving conflicts ended up being the responsibility of the release champion, who may or may not have all the context needed. With our cherry-picking flow, the conflicts end up having to be resolved more with more granularity the _other way around_.
+
+Developers fix their bugs on `main`, meaning the code they need to modify to address the issue has moved on from the base the release branch was cut from. If a fix can’t be cleanly cherry-picked onto the release branch, the author of the fix is prompted to create a pull request _per-fix_, addressing more granular conflicts as they come up!
+
+### Building a release process for portability
+Across M&S we have more than 100 mobile applications in their own individual repositories. A lot of these have their own, manual release process, and a consistent branching structure doesn’t exist. We knew this coming in to the reimagining of our releasing process, and built it from the ground up with _portability_ in mind.
+
+Our release process can be summarized as just two manual steps:
+1. Create the release branch, and wait for stability.
+2. Deploy the release to production.
+
+Along the way we have automated processes that kick in and do the rest of the work: Commits to release branches submit builds to the stores, and a special cherry-pick label manages fixes destined for the release.
+
+The common points across _all_ mobile release then end up being:
+1. Releases are cut from the `main` branch.
+2. Commits to the `release/X` branch builds the apps, and stages them somewhere.
```

Comment:
This is the first time you've mentioned `release/X` maybe reference that in the first of the bullet points above. It means that as the reader progresses through this section they are tying together concepts from higher up.

Reply:
"finalized" below could be elaborated on some more. As an observer of the work you guys did on the release process I think "finalizing" was a cool concept and I think readers could leave this post with the concept of "finalizing" and it could be something they attribute to this work

You could elaborate more on the comms, you could mention how on Android we have multiple app stores versus the on for Apple. You do touch on it below, but I think focus on finalization

Reply:
Good idea! I've reworked this section around these bullets to link branches together, and focus more on "finalizing" in general. Even made mention briefly of our producer/consumer model for reusable workflows, which I think fits well here!